docs/retention.md

```diff
@@ -0,0 +1,140 @@
+# Retention policy
+
+
+## Overview
```

Comment:
Most of the improvement suggestions I made in https://github.com/brainwane/zulip/commit/d70f8630769c0588f2ed2a0703110370265a4dc8 are still applicable to improve the clarity of this document - how about you integrate those (as long as I haven't introduced any inaccuracies) and then ping me so I can take a fresh look and help revise? Thanks!